edb/pgsql/compiler/pathctx.py

```diff
@@ -1087,6 +1087,19 @@ def has_type_rewrite(
     )
 
 
+def link_needs_type_rewrite(
+        typeref: irast.TypeRef, *, env: context.Environment) -> bool:
+    return (
+        has_type_rewrite(typeref, env=env)
+        # Typically we need to apply rewrites when looking at a link
+        # target that has a policy on it, but we suppress this for
+        # schema::ObjectType. None of the hidden objects should be
+        # user visible anyway, and this allows us to do type id
+        # injection without a join.
+        and str(typeref.real_material_type.name_hint) != 'schema::ObjectType'
```

Comment:
Comparing `name_hint`? Is lookup into the schema that slow?

Reply:
We don't have the schema in the ir->pg compiler

Reply:
Of course, yes